src/Umbraco.Core/Security/BackOfficeUserManager.cs

```diff
@@ -238,5 +237,244 @@ public override bool SupportsUserPhoneNumber
         /// Gets/sets the default back office user password checker
         /// </summary>
         public IBackOfficeUserPasswordChecker BackOfficeUserPasswordChecker { get; set; }
+
+        public override Task<IdentityResult> SetLockoutEndDateAsync(int userId, DateTimeOffset lockoutEnd)
+        {
+            var result = base.SetLockoutEndDateAsync(userId, lockoutEnd);
+
+            // The way we unlock is by setting the lockoutEnd date to the current datetime
+            if (result.Result.Succeeded && lockoutEnd >= DateTimeOffset.UtcNow)
+                RaiseAccountLockedEvent(userId);
+            else
+                RaiseAccountUnlockedEvent(userId);
+
+            return result;
+        }
+
+        public override Task<IdentityResult> AccessFailedAsync(int userId)
+        {
+            var result = base.AccessFailedAsync(userId);
+
+            //Slightly confusing: this will return a Success if we successfully update the AccessFailed count
+            if (result.Result.Succeeded)
+                RaiseLoginFailedEvent(userId);
+
+            return result;
+        }
+
+        public override Task<IdentityResult> ChangePasswordAsync(int userId, string currentPassword, string newPassword)
+        {
+            var result = base.ChangePasswordAsync(userId, currentPassword, newPassword);
+            if (result.Result.Succeeded)
+                RaisePasswordChangedEvent(userId);
+            return result;
+        }
+
+        public override async Task<IdentityResult> ResetAccessFailedCountAsync(int userId)
+        {
+            var user = ApplicationContext.Current.Services.UserService.GetUserById(userId);
+
+            if (user == null)
+            {
+                throw new ProviderException(string.Format("No user with the id {0} found", userId));
+            }
+
+            if (user.FailedPasswordAttempts > 0)
+            {
+                user.FailedPasswordAttempts = 0;
+                ApplicationContext.Current.Services.UserService.Save(user);
+                RaiseResetAccessFailedCountEvent(userId);
+            }
+
+            return await Task.FromResult(IdentityResult.Success);
+        }
+
+        /// <summary>
+        /// Clears a lock so that the membership user can be validated.
+        /// </summary>
+        /// <param name="username">The membership user to clear the lock status for.</param>
+        /// <returns>
+        /// true if the membership user was successfully unlocked; otherwise, false.
+        /// </returns>
+        public bool UnlockUser(string username)
```

Comment:
This seems quite odd to have here. It's called from `UmbracoMembershipProvider` only! I suppose that purely to keep all of these events in one place (i.e. the BackOfficeUserManager) but in that case this should be internal.  This logic has changed in 7.7 too and like the above comment, the BackOfficeUserManager shouldn't be referencing the UserService - it talks to the UserStore.

Reply:
Have pushed some notes for this and changed to internal and will need to change this logic for the 7.7 branch. I'll get that done when i merge